# HG changeset patch
# User John D <john.donoghue@ieee.org>
# Date 1505678029 14400
#      Sun Sep 17 15:53:49 2017 -0400
# Node ID f7710116bf7d4f50348a482a35c9e63b41212bae
# Parent  838fdcd4edf63dcf4c8569e0d31743abf3fbf8b5
qterminal(win32): mark null characters as spaces when copying to clipboard

* libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp:
  (QConsolePrivate::getSelection): if character is null, set it to a space character

diff --git a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
--- a/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
+++ b/libgui/qterminal/libqterminal/win32/QWinTerminalImpl.cpp
@@ -538,16 +538,18 @@ QString QConsolePrivate::getSelection (v
                 {
                   if (lastNonSpace >= 0)
                     selection.truncate (lastNonSpace);
                   selection.append ('\n');
                   lastNonSpace = selection.length ();
                 }
 
               QChar c (buf[i].Char.UnicodeChar);
+              if (c.isNull ())
+                c = QChar (QChar::Space);
 
               selection.append (c);
               if (! c.isSpace ())
                 lastNonSpace = selection.length ();
             }
 
           if (lastNonSpace >= 0)
             selection.truncate (lastNonSpace);
